addons/sale/migrations/12.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_renames = {
+    'payment_transaction': [
+        ('sale_order_id', None),
+    ],
+    'sale_order': [
+        ('payment_tx_id', None),
+    ],
+}
+
+_field_renames = [
+    ('sale.order.line', 'sale_order_line', 'amt_invoiced',
+     'untaxed_amount_invoiced'),
+    ('sale.order.line', 'sale_order_line', 'amt_to_invoice',
+     'untaxed_amount_to_invoice'),
+]
+
+_field_renames_order_dates = [
+    ('sale.order', 'sale_order', 'requested_date', 'commitment_date'),
+]
+
+
+def fill_sale_order_line_sections(cr):
+    """It's done here instead of post-migration to avoid
+    possible new rows added in the migration"""
+    cr.execute(
+        "ALTER TABLE sale_order_line ADD COLUMN display_type varchar",
+    )
+    openupgrade.logged_query(
+        cr, """
+        INSERT INTO sale_order_line (order_id, layout_category_id,
+            sequence, name, price_unit, product_uom_qty, customer_lead,
+            display_type, create_uid, create_date, write_uid, write_date)
+        SELECT sol.order_id, sol.layout_category_id,
+            min(sol.sequence) -1 as sequence, max(slc.name), 0, 0, 0,
+            'line_section', min(sol.create_uid), min(sol.create_date),
+            min(sol.write_uid), min(sol.write_date)
+        FROM sale_order_line sol
+        INNER JOIN sale_layout_category slc ON slc.id = sol.layout_category_id
+        GROUP BY order_id, layout_category_id
+        ORDER BY order_id, layout_category_id, sequence
+        """
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if openupgrade.column_exists(env.cr, 'sale_order', 'payment_tx_id'):
+        # from sale_payment module
+        openupgrade.rename_columns(env.cr, _column_renames)
+    openupgrade.rename_fields(env, _field_renames)
+    if openupgrade.column_exists(env.cr, 'sale_order', 'requested_date'):
+        # from sale_order_dates module
+        openupgrade.drop_columns(env.cr, [('sale_order', 'commitment_date')])
+        openupgrade.rename_fields(env, _field_renames_order_dates)
```

Comment:
@pedrobaeza I get an error here. Should I do a new `drop_fields` instead of the `drop_columns`? Or should I just do a `rename_columns` instead of this `drop_columns+rename_fields`? 🤔

Reply:
solved